source/Nuke.Common/EnvironmentInfo.Others.cs

```diff
@@ -23,7 +24,9 @@ public static partial class EnvironmentInfo
         public static string WorkingDirectory
         {
 #if NETCORE
-            get => Directory.GetCurrentDirectory();
+            get => ValidDirectoryPath(Directory.GetCurrentDirectory())
+                ? Directory.GetCurrentDirectory()
+                : Path.GetDirectoryName(Process.GetCurrentProcess().MainModule?.FileName);
```

Comment:
If a standard approach with valid path(not points to temp path) returns as previous otherwise withdraws path from the process.